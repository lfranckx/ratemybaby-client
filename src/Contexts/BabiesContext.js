import React, { Component } from 'react';

const BabiesContext = React.createContext({
    usersBabies: [],
    babies: [],
    error: null,
    setError: () => {},
    clearError: () => {},
    setBabies: () => {},
    setUsersBabies: () => {}
})

export default BabiesContext

export class BabiesProvider extends Component {
    state = {
        usersBabies: [],
        babies: [],
        error: null
    }
    
    setBabies = babies => {
        this.setState({ babies })
    }

    setUsersBabies = babies => {
        console.log('setUsersBabies:', babies);
        this.setState({ usersBabies: babies })
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    render() {
        const value = {
            babies: this.state.babies,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setBabies: this.setBabies,
            setUsersBabies: this.setUsersBabies
        }
        return (
            <BabiesContext.Provider value={value}>
                {this.props.children}
            </BabiesContext.Provider>
        )
    }
}