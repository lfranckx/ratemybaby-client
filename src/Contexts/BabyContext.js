import React, { Component } from 'react'
import BabyApiService from '../Services/baby-api-service'

export const nullBaby = {
    baby: null
}

const BabyContext = React.createContext({
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    updateBaby: () => {}
})

export default BabyContext

export class BabyProvider extends Component {
    state = {
        baby: nullBaby,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        console.log('running BabyContext clearError()');
        this.setState({ error: null })
    }

    setBaby = baby => {
        console.log('setting baby:', baby);
        this.setState({ 
            baby: baby 
        })
    }

    clearBaby = () => {
        console.log('running BabyContext clearBaby()');
        this.setBaby(nullBaby)
    }   

    updateBaby = baby => {
        console.log('updating baby:', baby, 
        'total_score:', baby.total_score,
        'total_votes:', baby.total_votes)
        this.setState({ 
            ...this.state.baby,
            baby 
        })
        BabyApiService.updateBaby(baby)
    }
      
    render() {
        console.log('BabyContext State', this.state);
        
        const value = {
            baby: this.state.baby,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setBaby: this.setBaby,
            clearBaby: this.clearBaby,
            updateBaby: this.updateBaby,
            clearUser: this.clearUser
        }
        return (
            <BabyContext.Provider value={value}>
                {this.props.children}
            </BabyContext.Provider>
        )
    }
}