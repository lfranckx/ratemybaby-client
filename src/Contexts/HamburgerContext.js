import React, { Component } from 'react'

export const nullBaby = {
    baby: null
}

const HamburgerContext = React.createContext({
    active: false,
    usersBabies: [],
    babies: [],
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    toggleActive: () => {},
    setNotActive: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    setUsersBabies: () => {}
})

export default HamburgerContext

export class HamburgerProvider extends Component {
    state = {
        active: false,
        usersBabies: [],
        babies: [],
        baby: nullBaby,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    toggleActive = () => {
        this.setState({ active: !this.state.active })
    }

    setNotActive = () => {
        this.setState({ active: false })
    }

    setBaby = baby => {
        this.setState({ 
            baby: baby 
        })
    }

    clearBaby = () => {
        this.setBaby(nullBaby)
    }   

    setUsersBabies = usersBabies => {
        this.setState({ usersBabies })
    }
      
    render() {
        const value = {
            active: this.state.active,
            usersBabies: this.state.usersBabies,
            babies: this.state.babies,
            baby: this.state.baby,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            toggleActive: this.toggleActive,
            setNotActive: this.setNotActive,
            setBabies: this.setBabies,
            setUsersBabies: this.setUsersBabies,
            setBaby: this.setBaby,
            clearBaby: this.clearBaby,
        }
        return (
            <HamburgerContext.Provider value={value}>
                {this.props.children}
            </HamburgerContext.Provider>
        )
    }
}