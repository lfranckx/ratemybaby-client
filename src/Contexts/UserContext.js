import React, { Component } from 'react'
// import UserApiService from '../Services/user-api-service'
import BabyApiService from '../Services/baby-api-service'

export const nullUser = {
    user: {}
}

const UserContext = React.createContext({
    user: nullUser,
    error: null,
    setError:() => {},
    clearError: () => {},
    setUser: () => {},
    clearUser: () => {},
    updateBaby: () => {},
    
})

export default UserContext

export class UserProvider extends Component {
    state = {
        user: nullUser,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setUser = user => {
        console.log('setting state from UserContext:', user);
        this.setState({ user })
    }

    clearUser = () => {
        this.setUser(nullUser)
    }   

    updateBaby = baby => {
        console.log('updating baby...', baby, 
        'total_score:', baby.total_score,
        'total_votes:', baby.total_votes)
        this.setState({ 
            ...this.state.baby,
            baby 
        })
        BabyApiService.updateBaby(baby)
    }

    render() {
        const value = {
            user: this.state.user,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setUser: this.setUser,
            clearUser: this.clearUser,
            updateBaby: this.updateBaby
        }
        return (
            <UserContext.Provider value={value}>
                {this.props.children}
            </UserContext.Provider>
        )
    }
}