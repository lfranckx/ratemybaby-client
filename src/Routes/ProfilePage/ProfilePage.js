import React, { Component } from 'react'
import BabyApiService from '../../Services/baby-api-service'
import BabyContext from '../../Contexts/BabyContext'
import UserProfile from '../../Components/UserProfile/UserProfile'

class ProfilePage extends Component {

    static defaultProps = {
        match: { params: {} }
    }
    
    static contextType = BabyContext
    
    componentDidMount() {
        this.context.clearError()
        this.context.setNotActive()
        let babyId = this.props.match.params.babyId
        BabyApiService.getBaby(babyId)
            .then(this.context.setBaby)
            .catch(this.context.setError)
    }
    
    componentWillUnmount() {
        this.context.clearError()
        BabyApiService.getByParentId()
            .then(res => {
                this.context.setUsersBabies(res)
            })
            .catch(this.context.setError)
    }
    
    render() {
        const { baby } = this.context
        
        if (!baby) {
            return <div className='loading'>Loading...</div>
        }
        return( 
            <UserProfile baby={baby}/>
        )
    }
}

export default ProfilePage