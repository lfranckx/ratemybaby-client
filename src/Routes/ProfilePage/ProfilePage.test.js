import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfilePage from './ProfilePage';
import { HamburgerProvider } from '../../Contexts/HamburgerContext';

describe(`ProfilePage component`, () => {

    it(`renders profile page without crashing`, () => {
        const wrapper = shallow(
            <HamburgerProvider>
                <ProfilePage />
            </HamburgerProvider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});