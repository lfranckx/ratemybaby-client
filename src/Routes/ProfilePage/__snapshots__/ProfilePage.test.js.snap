// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component renders profile page without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "active": false,
      "babies": Array [],
      "baby": Object {
        "baby": null,
      },
      "clearBaby": [Function],
      "clearError": [Function],
      "error": null,
      "setBabies": undefined,
      "setBaby": [Function],
      "setError": [Function],
      "setNotActive": [Function],
      "setUsersBabies": [Function],
      "toggleActive": [Function],
      "usersBabies": Array [],
    }
  }
>
  <ProfilePage
    match={
      Object {
        "params": Object {},
      }
    }
  />
</ContextProvider>
`;
